examples/progress.rs

```diff
@@ -0,0 +1,203 @@
+//!
+//! Clicking a button will open the webpage to that option.
+//!
+//! Clicking the clear button will clear the list box and send a command to the
+//! background that waits 2 seconds before issuing a reload command back to the
+//! component, which forwards the reload command back to the caller of the
+//! component, which then issues to reload the widgets again.
+
+use futures::FutureExt;
+use gtk::prelude::*;
+use relm4::*;
+
+fn main() {
+    gtk::Application::builder()
+        .application_id("org.relm4.ProgressExample")
+        .launch(|_app, window| {
+            // Intiialize a component's root widget
+            App::init()
+                // Attach the root widget to the given window.
+                .attach_to(&window)
+                // Start the component service with an initial parameter
+                .launch("Settings List Demo".into());
+        });
+}
+
+#[derive(Default)]
+pub struct App {
+    computing: bool,
+    task: Option<CmdOut>,
+}
+
+pub struct Widgets {
+    button: gtk::Button,
+    label: gtk::Label,
+    progress: gtk::ProgressBar,
+}
+
+pub enum Input {
+    Compute,
+}
+
+pub enum Output {
+    Clicked(u32),
+}
+
+pub enum Command {
+    Compute,
+}
+
+pub enum CmdOut {
+    Progress(f32),
+    Finished(Result<String, ()>),
+}
+
+impl Component for App {
+    type Command = Command;
+    type CommandOutput = CmdOut;
+    type Input = Input;
+    type Output = Output;
+    type InitParams = String;
+    type Root = gtk::Box;
+    type Widgets = Widgets;
+
+    fn init_root() -> Self::Root {
+        gtk::Box::builder()
+            .halign(gtk::Align::Center)
+            .valign(gtk::Align::Center)
+            .orientation(gtk::Orientation::Vertical)
+            .build()
+    }
+
+    fn init_parts(
+        _args: Self::InitParams,
+        root: &Self::Root,
+        input: &Sender<Self::Input>,
+        _output: &Sender<Self::Output>,
+    ) -> ComponentParts<Self, Self::Widgets> {
+        relm4_macros::view! {
+            container = gtk::Box {
+                set_width_request: 300,
+                set_spacing: 12,
+                set_orientation: gtk::Orientation::Horizontal,
+
+                append = &gtk::Box {
+                    set_spacing: 4,
+                    set_hexpand: true,
+                    set_valign: gtk::Align::Center,
+                    set_orientation: gtk::Orientation::Vertical,
+
+                    append: label = &gtk::Label {
+                        set_label: "Find the answer to life:",
+                    },
+
+                    append: progress = &gtk::ProgressBar {
+
+                    },
+                },
+
+                append: button = &gtk::Button {
+                    set_label: "Compute",
+                    connect_clicked(input) => move |_| {
+                        input.send(Input::Compute);
+                    }
+                }
+            }
+        }
+
+        root.append(&container);
+
+        progress.hide();
```

Comment:
`set_visible` in the view! macro would be shorter

Reply:
Indeed. And it's not necessary to use append anymore.